### PR TITLE
Pedestrian crossings

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
SUMMARY: Content "Pedestrian crossings."

#### Purpose of change
Continuation of making our cities slightly more realistic.

#### Describe the solution
Added `t_zebra` terrain and added it to mapdata.h and .cpp files. Made pedestrian crossings spawn on all intersections and with 10% chance - on ordinary two-way straight roads.

#### Describe alternatives you've considered
None.

#### Testing
Teleported around cities and checked the roads.

#### Additional context
Four-way intersection
изображение

Tee-shaped intersection
изображение

### CODE REVIEW

**Comment by Xaleth** [comment]
where did the stop lines go?

**Comment by Night-Pryanik** [comment]
Stop lines have only 50% chance to spawn on intersections. Here's the screenshot with stop lines
изображение

File: src/mapgen_functions.cpp [hunk 1]
```diff
@@ -899,27 +899,50 @@ void mapgen_road( mapgendata &dat )
         }
 
         // four-way intersections
-        if( num_dirs == 4 && one_in( 2 ) ) {
-            m->furn_set( point( 3, 3 ), f_traffic_light );
-            m->furn_set( point( 3, 20 ), f_traffic_light );
-            m->furn_set( point( 20, 3 ), f_traffic_light );
-            m->furn_set( point( 20, 20 ), f_traffic_light );
+        if( num_dirs == 4 ) {
+            square( m, t_zebra, point( 4, 1 ), point( 19, 2 ) );
```
Comment (8street) [inline src/mapgen_functions.cpp:903]:
```suggestion
            // No need to draw a zebra if there is no sidewalk nearby
            if (m->ter(point(3, 1)) == t_sidewalk && m->ter(point(20, 2)) == t_sidewalk) {
                square( m, t_zebra, point( 4, 1 ), point( 19, 2 ) );
            }
```
It should be copied for other 3 cases.

**Comment by Zireael07** [comment]
This is awesome, but the zebra on bottom part of the T kinda bugs me aesthetically. Can you use the | character instead of - for it (I seem to remember we have a way to use directional terrain/glyphs?)

File: data/json/furniture_and_terrain/terrain-floors-outdoors.json [hunk 2]
```diff
@@ -201,6 +201,25 @@
...
     }
   },
+  {
+    "type": "terrain",
+    "id": "t_zebra",
+    "alias": [ "t_zebra_h", "t_zebra_v" ],
+    "name": "pedestrian crossing",
+    "connects_to": "PAVEMENT",
+    "description": "White polymer lines painted on the pavement which indicate places where pedestrians should cross the road.",
+    "symbol": "-",
```
Comment (John-Candlebury) [inline data/json/furniture_and_terrain/terrain-floors-outdoors.json:211]:
```suggestion
    "symbol": ".",
```
The `-` kinda mkes them look like obstacles, I think it feels visually confusing.

**Comment by Xaleth** [comment]
At this point even with the graphics they look like they don't fit in. Maybe make the black background a little lighter so it can blend in more around the environment

**Comment by anothersimulacrum** [comment]
What graphics? Those are filler characters.

File: src/mapgen_functions.cpp (hunk 1 above)
Comment (8street) [reply src/mapgen_functions.cpp:903]: Also I'e saw google maps. Not every T intersection has all pedestrian crossings. And not every 4 way intersection has 2 crossings. Also, in some suburbs there are almost no pedestrian crossings. For example 1 for 5-10 intersections.

**Comment by mrkybe** [comment]
I think only every other tile should have the crossing texture, or at least there two should be two crossing textures and it swaps between them.  Looking at how crossings are patterned in Boston MA,
image
There's like 3-4 bars per lane of road.  Cata version will be 4x that, because there's already 2x the tiles per lane and each one will need both black and white on it.

File: src/mapgen_functions.cpp (hunk 1 above)
Comment (Night-Pryanik) [reply src/mapgen_functions.cpp:903]: I guess adding some randomness in placing crossings wouldn't hurt. Will get on this tomorrow.

**Comment by Zireael07** [comment]
How does the zebra look now?

**Comment by Night-Pryanik** [comment]
> How does the zebra look now?

I got two variants:
изображение

The one at the upper side is placed at every other tile.
The other ones are placed on each tile.

I can't decide which one is better.

(Graphics is temporary, don't pay much attention to it).

**Comment by Zireael07** [comment]
Every other (like at the upper side) looks better IMHO - the ones that stretch across the whole road without breaks might look like obstacles to inexperienced/confused players.

**Comment by Night-Pryanik** [comment]
All right, done.

**Pull request merged**